### PR TITLE
Add Librdkafka

### PR DESCRIPTION
**Pull request by wenhaocs**
Add librdkafka related staff to nebula, which is needed by rocksdb cloud.

### CODE REVIEW

File: project/CMakeLists.txt [hunk 1]
```diff
@@ -281,11 +287,12 @@ maybe_add_dependencies(fizz folly)
 maybe_add_dependencies(wangle folly fizz)
 maybe_add_dependencies(fbthrift folly krb5 bison flex mstch zlib zstd wangle fatal)
 maybe_add_dependencies(proxygen wangle libunwind gperf)
+maybe_add_dependencies(librdkafka berkeleydb libcurl libtool openssl cyrus-sasl)
```
Comment (panda-sheep) [inline project/CMakeLists.txt:290]: Maybe move this line after line 294 
so that `librdkafka` is only compiled when `ENABLE_ROCKSDB_CLOUD` is true.  WDYT?

**Review by panda-sheep — COMMENTED** [review]
Good job！

File: project/CMakeLists.txt (hunk 1 above)
Comment (wenhaocs) [reply project/CMakeLists.txt]: good catch

**Review by panda-sheep — APPROVED** [review]

File: project/externals/librdkafka.cmake [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ExternalProject_Add_Step(${name} install-static
+    DEPENDEES build
+    DEPENDERS install
+    ALWAYS false
+    COMMAND
+        make -s install -j${BUILDING_JOBS_NUM}
+    COMMAND
+        find ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR} -name "librdkafka*.so*" -delete
+    WORKING_DIRECTORY ${source_dir}
+)
```
[the remark this answers is not in the document]
Comment (wenhaocs) [reply project/externals/librdkafka.cmake:38]: I am not sure why `*.so` are still generated under this. The current status is `it works but not final`. I disabled `gssapi` which may be used when connecting to cloud based kafka. We will sure revisit this.

[the remark this answers is not in the document]
Comment (wenhaocs) [reply project/externals/librdkafka.cmake:38]: The current configure of `librdkafka` is used to compile `librdkafka` for connecting kafka without SALS, which meets the current need (we have no agenda to use Kafka in cloud yet). So yes, all the dependencies needed at the current stage have existed in our third party package.

**Pull request merged**